### PR TITLE
Use the standard credential provider chain to support IRSA

### PR DESCRIPTION
**Pull request by ellistarn**
*Issue #, if available:*

## Description 

Replace `ec2rolecreds` with the standard credential provider chain. This will use the following priorities: https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials

* Environment variables (IRSA)
* EC2 Metadata (IMDS)

Which supports different runtime environments where VPCRC is deployed.

## Extras
* Improved logging w/ error wrapping, as I was struggling to see what was going wrong
* Removed some unnecessary interfaces, that only had (and would ever have) a single implementation.

## Tested with

```
go test -v -timeout 60m ./test/integration/webhook -cluster-name dev -aws-region us-west-2 -cluster-kubeconfig /Users/etarn/.kube/config -aws-vpc-id vpc-0281c2e6397718daa
go test -v -timeout 60m ./test/integration/perpodsg -cluster-name dev -aws-region us-west-2 -cluster-kubeconfig /Users/etarn/.kube/config -aws-vpc-id vpc-0281c2e6397718daa
```

* EC2 Instance Creds (IMDS)
* IRSA

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/framework/resource/k8s/namespace/manager.go [hunk 1]
```diff
@@ -51,7 +52,7 @@ func (m *defaultManager) DeleteAndWaitTillNamespaceDeleted(ctx context.Context,
 	namespaceObj := &v1.Namespace{ObjectMeta: metaV1.ObjectMeta{Name: namespace, Namespace: ""}}
 	err := m.k8sClient.Delete(ctx, namespaceObj)
 	if err != nil {
-		return err
+		return client.IgnoreNotFound(err)
```
Comment (haouc) [inline test/framework/resource/k8s/namespace/manager.go:55]: Thanks. Seems our test didn't cover this use case...

File: test/framework/resource/k8s/pod/manager.go [hunk 2]
```diff
@@ -119,14 +119,13 @@ func (d *defaultManager) GetPodsWithLabel(context context.Context, namespace str
...
-	if err != nil {
-		return err
+	if err := d.k8sClient.Delete(context, pod); err != nil {
+		return client.IgnoreNotFound(err)
 	}
 
 	observedPod := &v1.Pod{}
 	return wait.PollUntil(utils.PollIntervalShort, func() (done bool, err error) {
-		err = d.k8sClient.Get(context, utils.NamespacedName(pod), observedPod)
+		err = d.k8sClient.Get(context, client.ObjectKeyFromObject(pod), observedPod)
```
Comment (haouc) [inline test/framework/resource/k8s/pod/manager.go:128]: Nice. I think this is the better way to retrieve the key.

File: test/integration/perpodsg/perpodsg_suite_test.go [hunk 3]
```diff
@@ -56,14 +55,8 @@ var _ = BeforeSuite(func() {
...
 var _ = AfterSuite(func() {
-	By("waiting for all the ENIs to be deleted after being cooled down")
-	time.Sleep(time.Second * 90)
-
-	err := frameWork.EC2Manager.DeleteSecurityGroup(securityGroupID1)
-	Expect(err).ToNot(HaveOccurred())
-
-	frameWork.EC2Manager.DeleteSecurityGroup(securityGroupID2)
-	Expect(err).ToNot(HaveOccurred())
+	Expect(frameWork.EC2Manager.DeleteSecurityGroup(ctx, securityGroupID1)).To(Succeed())
```
Comment (haouc) [inline test/integration/perpodsg/perpodsg_suite_test.go:58]: I like `To(Succeed())` 👍

**Review by haouc — COMMENTED** [review]
Thanks for making the code cleaner. Just couple of concerns.

File: test/integration/perpodsg/perpodsg_suite_test.go (hunk 3 above)
Comment (haouc) [reply test/integration/perpodsg/perpodsg_suite_test.go:58]: @ellistarn ran some tests from control plane as well. This change looks good to me. Could you check if you need clean up the PR as commented above before we can merge? Thanks.

File: test/framework/resource/aws/ec2/manager.go [hunk 4]
```diff
@@ -125,14 +111,20 @@ func (d *defaultManager) RevokeSecurityGroupIngress(securityGroupID string, port
...
 
-func (d *defaultManager) DeleteSecurityGroup(securityGroupID string) error {
-	_, err := d.ec2Client.DeleteSecurityGroup(&ec2.DeleteSecurityGroupInput{
-		GroupId: &securityGroupID,
-	})
-	return err
+func (d *Manager) DeleteSecurityGroup(ctx context.Context, securityGroupID string) error {
+	return wait.PollUntil(utils.PollIntervalShort, func() (done bool, err error) {
+		if _, err = d.ec2Client.DeleteSecurityGroup(&ec2.DeleteSecurityGroupInput{GroupId: &securityGroupID}); err != nil {
+			if err.(awserr.Error).Code() == "DependencyViolation" {
```
[the remark this answers is not in the document]
Comment (ellistarn) [reply test/framework/resource/aws/ec2/manager.go:117]: If stuck, this should us the standard test timeout. Generally, I'd avoid wait.PollUntil, and use ginkgo's `Eventually`. Do you want to address in a followon or want me to do this now?

[the remark this answers is not in the document]
Comment (haouc) [reply test/framework/resource/aws/ec2/manager.go:117]: Thanks, Ellis. I think we can do as a follow-up.

**Comment by haouc** [comment]
lgtm

**Review by haouc — APPROVED** [review]

**Pull request merged**